lib/run_cucumber.js

```diff
@@ -6,64 +6,91 @@ const session_vars = require('./utils/session_vars');
-  }
-
-
-  // If they're running it from an interview on their server instead of from GitHub
-  if ( session_vars.get_origin() === `playground` ) {
-    provided_config.paths.push(`/usr/share/docassemble/files/playgroundsources/${session_vars.get_user_id()}/${session_vars.get_user_project_name()}/*.feature`);
-    // For S3
-    provided_config.paths.push(`/tmp/playgroundsources/${session_vars.get_user_id()}/${session_vars.get_user_project_name()}/*.feature`);
+function build_config( command_args ) {
+  // Turn all the paths to the sources folders into paths to the feature files
+  let sources_paths = session_vars.get_sources_paths();
+  let feature_paths = [];
+  for ( let source of sources_paths ) {
+    feature_paths.push( `${ source }/*.feature` );
   }
 
-  let tags = argv.slice(2);
-  if ( session_vars.get_origin() === `playground` ) {
-    tags = session_vars.get_tags();
+  const config = {
+    publishQuiet: true,
+    require: `./lib/index.js`,
+    format: [`progress`, `summary:cucumber-report.txt`],
+    paths: feature_paths,
+    retry: 1,
   }
 
+  // Normalize tag expressions, if there are any
+  let tags = command_args[`_`];
   if (tags.length > 0) {
     let joined_tags = '';
     // Cucumber expects tags in boolean expressions, like (@a0 or @a1) and @random.
     // If we get a list of only tags, then "or" them all together
-    if (tags.every((val) => val.startsWith("@"))) {
-      joined_tags = tags.join(' or ');
+    if (tags.every((val) => val.startsWith(`@`))) {
+      joined_tags = tags.join(` or `);
     }
     // If we get a list of seemingly no tags, assume they meant to use tags
-    else if (tags.every((val) => !val.startsWith("@"))) {
-      joined_tags = tags.map((val) => "@" + val).join(' or ');
+    else if (tags.every((val) => !val.startsWith(`@`))) {
+      joined_tags = tags.map((val) => `@` + val).join(` or `);
     }
     // If it's a mix of tags and no tags, it's probably a boolean expression!
     else {
-      joined_tags = tags.join(' ');
+      joined_tags = tags.join(` `);
     }
 
-    provided_config.tags = joined_tags;
-  }  // end if there are tags
+    config.tags = joined_tags;
+  }
+
+  return config;
+};  // Ends build_config()
 
+
+(async function main() {
+  /** Create the cucumber config, run the tests, create and
+  *   manipluate the output. */
+  const environment = { cwd: process.cwd() };
+  console.log("alkiln-run: environment: %o", environment);
+
+  // process.argv is a list of strings of commands and args.
+  // If using `npm run cucumber`, use `--` before `--sources=./foo`
+  let argv = require(`minimist`)( process.argv.slice(2), {
+    default: {
+      // Testing from GitHub. TODO: Do we want to allow multiple sources paths?
+      sources: [ `./docassemble/*/data/sources` ],
+      // From playground: `/usr/share/docassemble/files/playgroundsources/<user_id>/<project>/*.feature`
+      // From playground with S3: `/tmp/playgroundsources/<user_id>/<project>/*.feature`
+    }
+  });
```

Comment:
Do you mean like this?

```suggestion

  // We'll use the user info to construct the possible paths to /sources
  let id = session_vars.get_user_id();
  // `project` is the name in the server docassemble "Project" page
  let project = session_vars.get_user_project_name();

  // process.argv is a list of strings of commands and args.
  // If using `npm run cucumber`, use `--` before `--sources=./foo`
  let argv = require(`minimist`)( process.argv.slice(2), {
    default: {
      sources: [
        `./docassemble/*/data/sources`,  // In GitHub folder
        `/usr/share/docassemble/files/playgroundsources/${id}/${project}/*.feature`, // From playground without S3
        `/tmp/playgroundsources/${id}/${project}/*.feature` // From playground with S3
      ],
    }
  });
```

Reply:
Yeah! I'd make sure the tests run fine, we can't really test this because it's not gonna be used in ALKilnInThePlayground, but it can't hurt, I like it a lot better than commented code.